framework/widgets/DetailView.php

```diff
@@ -231,6 +231,10 @@ protected function normalizeAttributes()
                 throw new InvalidConfigException('The attribute configuration requires the "attribute" element to determine the value and display label.');
             }
 
+            if ($attribute['value'] instanceof \Closure) {
+                $attribute['value'] = call_user_func($attribute['value'], $this->model);
```

Comment:
Good idea. That would be consistent with `DataColumn` value callback: `function ($model, $key, $index, $column)`.